src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Can't add this vale");
+        }
+        if (elementData.length == size) {
+            elementData = grow();
+        }
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index > size - 1) {
+            throw new ArrayListIndexOutOfBoundsException("Incorrect index");
+        }
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < 0 || index > size - 1) {
+            throw new ArrayListIndexOutOfBoundsException("Incorrect index");
+        }
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index > size - 1) {
+            throw new ArrayListIndexOutOfBoundsException("Incorrect index");
```

Comment:
```suggestion
            throw new ArrayListIndexOutOfBoundsException("Incorrect index");
            throw new ArrayListIndexOutOfBoundsException("Incorrect index: " + index);
```
Думаю что во всех trow Exception нужно указывать какой именно index, value 
Что-то типа такого